rasterio/_base.pyx

```diff
@@ -38,70 +40,74 @@ def check_gdal_version(major, minor):
 
 def gdal_version():
     """Return the version as a major.minor.patchlevel string."""
-    #cdef const char *releasename = _gdal.GDALVersionInfo("RELEASE_NAME")
-    releasename = <bytes> _gdal.GDALVersionInfo("RELEASE_NAME")
-    return releasename.decode('utf-8')
+    return _gdal.GDALVersionInfo("RELEASE_NAME")
+
+
+cdef get_driver_name(GDALDriverH driver):
+    """Return Python name of the driver"""
+    return _gdal.GDALGetDriverShortName(driver)
 
 
 def get_dataset_driver(path):
     """Return the name of the driver that would be used to open the
     dataset at the given path."""
-    cdef GDALDatasetH hds = NULL
-    cdef GDALDriverH drv = NULL
-    cdef const char *drivername = NULL
-    cdef const char *filename = NULL
+    cdef GDALDatasetH dataset = NULL
+    cdef GDALDriverH driver = NULL
 
     path, archive, scheme = parse_path(path)
```

Comment:
@sgillies lines 57-59 could collapse down to `path = vsi_path(*parse_path(path)).encode('utf-8')`